### ISSUE TITLE
Makemigrations is creating django-celery-beat migration file.

### ISSUE THREAD

**Issue by aroraprince** [celery/django-celery-beat#406]
### Summary:

After adding django-celery-beat to my installed apps, if you run `python manage.py makemigrations`, it is creating a migration file in django-celery-beat installation directory.

* Celery Version: `celery==5.0.5`
* Celery-Beat Version: `django-celery-beat==2.2.0`

### Exact steps to reproduce the issue:
1. Added django-celery-beat to installed apps
2. ran `python manage.py makemigrations`

### Detailed information
OS: macOS Big Sur
Python version: 3.9.0, installed via pyenv.

Screenshot link: https://www.dropbox.com/s/d119blepz3nin1i/Screenshot%202021-05-06%20at%209.33.36%20PM.png?dl=0

Ideally no migrations file should be created locally.

**Comment by auvipy** [comment]
if it is an auto migration without any changes then it was somehow missed from git commit and should be in the package distribution!

**Comment by aroraprince** [comment]
Yes, it was an auto migration, withou any sort of code change in source.
I can pick this up if needed, seems a pretty small change.

**Comment by auvipy** [comment]
please send a PR!

**Comment by aroraprince** [comment]
Okay, will do by 10th May, most probably sooner,  I see the contribution guidelines are mentioned in the celery project.

**Comment by aroraprince** [comment]
I cloned the repo and tried running `python manage.py makemigrations`, it is not creating any migration file in that case.
I was using `Django==3.2.1`, tried with that still the same.

Seems I am missing something here.

**Comment by auvipy** [comment]
what does python manage.py migrate do?

**Comment by aroraprince** [comment]
It ran the migration successfully till the 15th migration, without any warning that app is missing a migration to be created.
Since it is a different virtualenv, there is no way the previously created migration file, (no 16 as mentioned in screenshot) will cause any conflict.

I see Django has something new in settings now:
`DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'`

This could be causing that migration to be created.

**Comment by aroraprince** [comment]
Django comes with this as default now:

https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys

So this can cause confusion in a lot of projects, this may need some graceful way to handle.
If we explicitly set the datatype of id fields, it should solve the pupose.

**Comment by auvipy** [comment]
so it is a project-specific issue not to this package. we are not going to change the default value now.

**Comment by aroraprince** [comment]
I understand why it may be called a project-specific issue.
I am pretty sure since by default settings has this `DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'` in the new versions of Django.
This will happen in most of the cases when Django > 3.2 is used, if it is not updated again in future releases.

This can wait till more people report this.

**Comment by auvipy** [comment]
When defining a model, if no field in a model is defined with primary_key=True an implicit primary key is added. The type of this implicit primary key can now be controlled via the DEFAULT_AUTO_FIELD setting and AppConfig.default_auto_field attribute. No more needing to override primary keys in all models.

Maintaining the historical behavior, the default value for DEFAULT_AUTO_FIELD is AutoField. Starting with 3.2 new projects are generated with DEFAULT_AUTO_FIELD set to BigAutoField. Also, new apps are generated with AppConfig.default_auto_field set to BigAutoField. In a future Django release the default value of DEFAULT_AUTO_FIELD will be changed to BigAutoField.

To avoid unwanted migrations in the future, either explicitly set DEFAULT_AUTO_FIELD to AutoField:

**Comment by aotella** [comment]
Hi
I am facing similar issues. Migrations are being created locally and dependency is being added in the migrations file. When I switch to a different environment I get
`
django.db.migrations.exceptions.NodeNotFoundError: Migration Project.0006_auto_20210524_1048 dependencies reference nonexistent parent node ('django_celery_beat', '0016_auto_20210510_2155')`

Did you by any chance find a fix for it.

**Comment by auvipy** [comment]
> Hi was the above comment to add more context?
> Do you see this as a valid bug?

https://github.com/celery/django-celery-beat/blob/master/django_celery_beat/apps.py#L14 is already in the master just need a new release

**Comment by DaticsAI-Haseeb** [comment]
any update on this?
i ran into this problem and had to work around another way. i am running my project in a docker environment. when i run makemigrations, the 0016 migration asks for permission to write in the folder:
"/usr/local/lib/python3.8/site-packages/django_celery_beat/migrations/"

i added the permissions like this in the Dockerfile to fix it:
RUN chown -R app:app /usr/local/lib/python3.8/site-packages/django_celery_beat/migrations/

this seems to be a workaround and should come with the release.

**Comment by auvipy** [comment]
> any update on this? i ran into this problem and had to work around another way. i am running my project in a docker environment. when i run makemigrations, the 0016 migration asks for permission to write in the folder: "/usr/local/lib/python3.8/site-packages/django_celery_beat/migrations/"
> 
> i added the permissions like this in the Dockerfile to fix it: RUN chown -R app:app /usr/local/lib/python3.8/site-packages/django_celery_beat/migrations/
> 
> this seems to be a workaround and should come with the release.

I think you are facing another problem

**Comment by DaticsAI-Haseeb** [comment]
the issue was when i tried to makemigrations, it did not have the file last migration file and it tried to create it, but did not have the permission. so im saying that that 0016 migration should be in the release. or better yet the "https://github.com/celery/django-celery-beat/blob/master/django_celery_beat/apps.py#L14" should be released

> Migrations for 'django_celery_beat':
  /usr/local/lib/python3.8/site-packages/django_celery_beat/migrations/0016_auto_20211005_1108.py
    - Alter field id on clockedschedule
    - Alter field id on crontabschedule
    - Alter field id on intervalschedule
    - Alter field id on periodictask
    - Alter field id on solarschedule
Traceback (most recent call last):
  File "manage.py", line 23, in <module>
    main()
  File "manage.py", line 19, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/makemigrations.py", line 190, in handle
    self.write_migration_files(changes)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/makemigrations.py", line 228, in write_migration_files
    with open(writer.path, "w", encoding='utf-8') as fh:
PermissionError: [Errno 13] Permission denied: '/usr/local/lib/python3.8/site-packages/django_celery_beat/migrations/0016_auto_20211005_1108.py

**Comment by auvipy** [comment]
that;s already released on 2.2.1

**Comment by kakoma** [comment]
As of this writing, using **django 4.2** and the latest **django-celery-beat (2.0.5),** adding `DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'` to `settings.py` will result in migrations being generated when you run `python manage.py makemigrations`, which is problematic.

To get round this, one approach is to set the `BigAutoField` in your individual apps:

```
# Your app config file
from django.apps import AppConfig


class YourAppConfig(AppConfig):
    name = "nameofyourapp"
    default_auto_field = 'django.db.models.BigAutoField'
```

When you run `python manage.py makemigrations --check`, you'll still get:
```
WARNINGS:
django_celery_beat.ClockedSchedule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
	HINT: Configure the DEFAULT_AUTO_FIELD setting or the BeatConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
django_celery_beat.CrontabSchedule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
	HINT: Configure the DEFAULT_AUTO_FIELD setting or the BeatConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
django_celery_beat.IntervalSchedule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
	HINT: Configure the DEFAULT_AUTO_FIELD setting or the BeatConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
django_celery_beat.PeriodicTask: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
	HINT: Configure the DEFAULT_AUTO_FIELD setting or the BeatConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
django_celery_beat.SolarSchedule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
	HINT: Configure the DEFAULT_AUTO_FIELD setting or the BeatConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
No changes detected
```

But that's bearable.
